reconcile/utils/terrascript_client.py

```diff
@@ -2405,6 +2409,71 @@ def populate_tf_resource_s3_cloudfront(self, resource, namespace_info):
         bucket_policy_tf_resource = aws_s3_bucket_policy(identifier, **values)
         tf_resources.append(bucket_policy_tf_resource)
 
+        # aws_s3_bucket_acl
+        values = {}
+        values['name'] = identifier
+        values['tags'] = common_values['tags']
+
+        aws_s3_bucket_acl_resource = aws_s3_bucket_acl(identifier, **values)
+        tf_resources.append(aws_s3_bucket_acl_resource)
+
+        access_control_policy = {
+            'grant': {
+                'grantee': {
+                    'id': '${' + aws_s3_bucket_acl_resource.id + '}',
+                    'type': '${' + aws_s3_bucket_acl_resource.type + '}'
+                },
+                'permission': 'READ'
+            }
+        }
+        grant = {
+            'grantee': '${' + aws_s3_bucket_acl_resource.grantee + '}',
+            'permission': '${' + aws_s3_bucket_acl_resource.permission + '}'
+        }
+        owner = {
+            'id': '${' + aws_s3_bucket_acl_resource.id + '}'
+        }
+
+        values['access_control_policy'] = access_control_policy
+        values['grant'] = grant
+        values['owner'] = owner
+
+        aws_s3_bucket_acl_resource = aws_s3_bucket_acl(identifier, **values)
+        tf_resources.append(aws_s3_bucket_acl_resource)
+
+        output_name_0_13 = output_prefix + '__acl'
+        output_value = '${' + aws_s3_bucket_acl.acl + '}'
+        tf_resources.append(Output(output_name_0_13, value=output_value))
+
+        # bucket policy for standard logging to cloudfront
+        values = {}
+        values['bucket'] = identifier
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "Grant access to CloudFront Origin Identity",
+                    "Effect": "Allow",
+                    "Principal": {
+                        "CanonicalUser": "${" + aws_s3_bucket_acl.acl + "}"
+                    },
+                    "Action": "s3:*",
+                    "Resource":
+                        [f"arn:aws:s3:::{identifier}/{enable_dir}/*"
+                         for enable_dir
+                         in common_values.get('get_object_enable_dirs', [])]
+                }
+            ]
+        }
+        values['policy'] = json.dumps(policy, sort_keys=True)
+        values['depends_on'] = self.get_dependencies([bucket_tf_resource])
+        region = common_values.get('region') or \
+            self.default_regions.get(account)
+        if self._multiregion_account(account):
+            values['provider'] = 'aws.' + region
+        bucket_policy_tf_resource = aws_s3_bucket_policy(identifier, **values)
+        tf_resources.append(bucket_policy_tf_resource)
+
         # cloud front distribution
         values = common_values.get('distribution_config', {})
```

Comment:
with the current changes, every cloudfront distribution will define s3 logging.
i believe what we would want is to get the logging config, and only if it exists we add resources? that would also tell us the name of the bucket to log to.

Reply:
That would make sense, it would then look like only quay is affected with this at the moment until others decide to include the logging config. Regarding if the bucket policy is needed or not, would our automation be able to acknowledge the ACL if we don't specify the bucket policy within this PR?